src/materials/Material.js

```diff
@@ -86,6 +86,15 @@ Object.assign( Material.prototype, EventDispatcher.prototype, {
 
 			}
 
+			// for backward compatability if shading is set in the constructor
+			if ( key === 'shading' ) {
+
+				console.warn( 'THREE.' + this.type + ': .shading has been removed. Use the boolean .flatShading instead.' );
+				this.flatShading = ( newValue === FlatShading ) ? true : false;
```

Comment:
@mrdoob It appears that the `THREE.Legacy` setters/getters do not work as expected if the legacy property is set in the constructor. In hindsight, this makes sense since the instance does not exist yet.

I had to add a work-around here that I think is a bit hacky. Perhaps there is another solution... 

Or, perhaps you don't feel this is that hacky...